File: motor/motor_simulation.py
import motor

setpoint = 1
output = 0
prev_setpoint = 0
prev_output = 0

f = open("motor_simulation.csv", "w")
f.write("time,setpoint,speed,position\n")

for i in range(1000):
    prev_setpoint = setpoint
    setpoint = 1

    prev_output = output
    output = motor.motor(setpoint, prev_setpoint, prev_output)

    f.write("%.8f,%.8f,%.8f\n" % (i * motor.TS, setpoint, output))

f.close()

File: motor/motor_simulation.c
#include <stdlib.h>
#include <stdio.h>
#include "motor.h"

int main()
{
    float setpoint = 0, output = 0, prev_setpoint = 0, prev_output = 0, position = 0;
    FILE *file = fopen("motor_simulation.csv", "w");

    fprintf(file, "time,setpoint,output,position\n");

    for (int i = 0; i < 1000; i++)
    {
        setpoint = 1;
        output = motor(setpoint, prev_setpoint, prev_output);
        position += output * TS;

        fprintf(file, "%.8f,%.8f,%.8f,%.8f\n", TS * i, setpoint, output, position);

        prev_setpoint = setpoint;
        prev_output = output;
    }

    fclose(file);

    return 0;
}